backend/cellguide/api/common/data.py

```diff
@@ -0,0 +1,68 @@
+import gzip
+import json
+import os
+from collections import defaultdict
+
+from backend.cellguide.common.config import CellGuideConfig
+from backend.cellguide.common.constants import COMPUTATIONAL_MARKER_GENES_FOLDERNAME, MARKER_GENE_PRESENCE_FILENAME
+from backend.cellguide.common.providers.s3_provider import S3Provider
+from backend.cellguide.common.utils import get_object_key
+
+
+def _defaultdict_to_dict(d):
+    if isinstance(d, defaultdict):
+        # Convert the defaultdict to a dict and recursively apply this function
+        return {key: _defaultdict_to_dict(value) for key, value in d.items()}
+    else:
+        return d
+
+
+def _initialize_cellguide_marker_gene_dict():
+    bucket = CellGuideConfig().bucket
+    s3_provider = S3Provider()
+
+    latest_snapshot_identifier = (
+        s3_provider.download_file(bucket_name=bucket, object_key=get_object_key(object="latest_snapshot_identifier"))
```

Comment:
Discussion - does it make sense to add some error handling in case the snapshot download fails? I know there's some handling by the provider, but is it worth it to put this behind a try-catch and have some more graceful handling in your endpoint?

Reply:
We don't actually want graceful handling of the error - the endpoint should throw to indicate that it must be retried. Once the marker gene data cache is warmed up, then this entire codepath is skipped anyway.

Reply:
https://github.com/chanzuckerberg/single-cell-data-portal/blob/b3ffb32476502277bee243e64e74f949fcfb62f9/backend/cellguide/common/providers/s3_provider.py#L67

This will raise an error in the endpoint